app/components/ui/Datatable.tsx

```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+
+interface DatatableProps<T = any> {
+  rows: T[];
+  rowRenderer: (row: T) => JSX.Element;
+}
+
+export const Datatable = ({ rows, rowRenderer }: DatatableProps) => (
```

Comment:
You should also define the type parameter on `Datatable` as well so that type inference at the call site can propagate into the type parameter.

Type parameters on arrow functions are pretty funky in JSX, since there's ambiguity in the parser as to whether to interpret it as an opening HTML tag or a type parameter, but you can work around that by using the `<T extends unknown>` [trick](https://stackoverflow.com/a/45576880), or you can just use a normal `function`-keyword function.

```suggestion
export const Datatable = <T extends unknown>({ rows, rowRenderer }: DatatableProps<T>) => (
```

By defining `Datatable` this way, TypeScript can warn you if you pass in a `rowRenderer` that doesn't actually match the `rows`. This is actually super neat, because you don't even need to explicitly tell TypeScript the concrete type at the call site, since it's just completely inferred via the type of `rows`.

Also, I'd get rid of the default value of the type parameter on `DatatableProps`, just to force users of the type to have to explicitly pass it in (or better yet, explicitly propagate the type from elsewhere).